packages/js/product-editor/src/components/details-categories-field/create-category-modal.tsx

```diff
@@ -63,6 +53,9 @@ export const CreateCategoryModal: React.FC< CreateCategoryModalProps > = ( {
 	const [ categoryParent, setCategoryParent ] =
 		useState< ProductCategoryNode | null >( null );
 
+	const [ categoryParentTypedValue, setCategoryParentTypedValue ] =
+		useState< string >( '' );
```

Comment:
I don't think you *need* the `string` type here, as it should infer that from the `''` used as the default. But, it's okay to be explicit if you prefer.